src/amo/components/MastHead.js

```diff
@@ -34,10 +33,12 @@ export class MastHeadBase extends React.Component {
 
     return (
       <div className="MastHead">
-        <div id="tabzilla">
-          <a href="https://www.mozilla.org">Mozilla</a>
+        <div className="MastHead-top-row">
+          <div id="tabzilla">
+            <a href="https://www.mozilla.org">Mozilla</a>
+          </div>
+          <AuthenticateButton className="MastHead-auth-button" size="small" location={location} />
```

Comment:
It looks like the `router` has two methods that look like they might work, `createLocation()` and `createHref()`. Calling these without any arguments results in `TypeError: location is undefined` so I don't think they will work :(